### PR TITLE
GSoC 2024 NVMM: Third blog post

### PR DESCRIPTION
**Pull request by dalmemail**

### CODE REVIEW

File: content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+and compile. It doesn't work, sadly, something breaks after executing a few chunks of any given VM.
+
+However, I didn't make any progress on the SVM backend (which is still at a very early state). I'm beginning to realise that this is one thing I won't be able to finish during the coding period.
+I probably have all the pieces needed to port it but I have to look at the differences between SVM and VMX (there are a few) and that will probably take some time I might not have in the end.
+Also, it's worth to mention that despite getting more and more stuff working [the list of TODO things keeps getting longer and longer](https://github.com/dalmemail/haiku-nvmm/issues).
+Things on that list are improvements on performance and behavior (NVMM's kidnaps the CPU which is probably not the best behavior if we want the system to be responsive, ...). None of them is very long or difficult but there are a lot of those little things to do so they will end up requiring a considerable amount of time.
+
+Last week I shared a screenshot on the forum of the first ever VM to be virtualized:
+
+![The communication that occurs between the three processes as WebProce](/files/blog/dalme/vm.png)
```
Comment (waddlesplash) [inline content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md:30]: Incorrect image description?

File: content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## What's the current blocker?
+
+After compiling QEMU with NVMM support (which was hardest than I expected) I tried to run a VM on it. Of course it didn't work.
+The problem is, this doesn't happen right at the beginning of VM execution but after 10-or-so memory exits correctly handled by NVMM. So that makes it hard to debug because the problem isn't on the point it fails but before. It doesn't help that the VM runs on a "black box" mode in which you can't see what's happening inside. I've been reading Intel SDM Volume 3 (which covers VMX) and I'll probably be able to make some ugly hack to trigger VM exits after executing a single instruction. I've been trying some other indirect less-time-consuming strategies for the last two days without much luck, so I'll probably have to make those hacks on NVMM to be able to see what's going on.
```
Comment (waddlesplash) [inline content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md:35]: Why not just build QEMU in debug mode and debug QEMU itself?

File: content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
++++
+type = "blog"
+title = "[GSoC 2024] Hardware Virtualization: Progress Report #2"
+author = "dalme"
+date = "2024-07-24 19:59:09+02:00"
+tags = ["haiku", "software"]
++++
+
+Another month has gone by and we're entering now the last month of GSoC! So, what did I got done this month?
```
Comment (scottmc) [inline content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md:9]: should be get not got here

File: content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+Also, it's worth to mention that despite getting more and more stuff working [the list of TODO things keeps getting longer and longer](https://github.com/dalmemail/haiku-nvmm/issues).
+Things on that list are improvements on performance and behavior (NVMM's kidnaps the CPU which is probably not the best behavior if we want the system to be responsive, ...). None of them is very long or difficult but there are a lot of those little things to do so they will end up requiring a considerable amount of time.
+
+Last week I shared a screenshot on the forum of the first ever VM to be virtualized:
+
+![The communication that occurs between the three processes as WebProce](/files/blog/dalme/vm.png)
+
+## What's the current blocker?
+
+After compiling QEMU with NVMM support (which was hardest than I expected) I tried to run a VM on it. Of course it didn't work.
```
Comment (scottmc) [inline content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md:34]: should be harder instead of hardest here

Comment (dalmemail) [reply content/blog/dalme/2024-07-24_[gsoc_2024]_hardware_virtualization_progress_report_2.md]: Yes. One day I'll learn how to write. Thanks !

**Pull request merged**